### PR TITLE
Prevent transferredCanvasNames is not iterable (fixes #17751)

### LINKED ISSUE #17751: library_pthread.js: transferredCanvasNames is not iterable
**Version of emscripten/emsdk:** 3.1.20

Looks like https://github.com/emscripten-core/emscripten/pull/17577 (3.1.20) broke [library_pthread.js#L651](https://github.com/emscripten-core/emscripten/blob/3.1.20/src/library_pthread.js#L651)

---

With pthreads/OffscreenCanvas `transferredCanvasNames` can (and also will) be `0`. 

For example when creating a ProxyWorker on the main thread (`new emscripten::ProxyWorker();` @ `RuntimeEmscriptenOffscreen.cpp:107` in the stack below)

````
runtime-project-offscreen.runtime-multi-debug.js:4663 Uncaught (in promise) TypeError: transferredCanvasNames is not iterable
    at ___pthread_create_js (runtime-project-offscreen.runtime-multi-debug.js:4663:24)
    at __pthread_create (pthread_create.c:197)
    at std::__2::__libcpp_thread_create(unsigned long*, void* (*)(void*), std::__2::__libcpp_thread_create(unsigned long*, void* (*)(void*), void*) (__threading_support:375)
    at invoke_iiii (runtime-project-offscreen.runtime-multi-debug.js:16274:36)
    at std::__2::thread::thread<emscripten::ProxyWorker::ProxyWorker()::'lambda'(), std::__2::thread::thread<emscripten::ProxyWorker::ProxyWorker()::'lambda'(), void>(emscripten::ProxyWorker::ProxyWorker()::'lambda'()&&) (thread:309)
    at invoke_iii (runtime-project-offscreen.runtime-multi-debug.js:16230:36)
    at emscripten::ProxyWorker::ProxyWorker() (ProxyWorker.hpp:27)
    at invoke_ii (runtime-project-offscreen.runtime-multi-debug.js:16252:36)
    at engineemscriptenoffscreen::RuntimeEmscriptenOffscreen::run() (RuntimeEmscriptenOffscreen.cpp:107)
    at run_project_offscreen(std::__2::basic_string<char, std::__2::char_traits<char>, run_project_offscreen(std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >) (runtime-project-offscreen.cpp:17)
````

---

**Cause**: The change from [#17577](https://github.com/emscripten-core/emscripten/pull/17577/files#diff-31ce84d9e9748b123befe225a55c67718b58011f9b0a099c821e5c5f0019b1faR651) will handle this case differently:

**[in] transferredCanvasNames**
```
const transferredCanvasNames = 0;
for (let i in transferredCanvasNames) {
  console.log(i);
}
````
=>will just do nothing (skips the for-loop)

**[of] transferredCanvasNames**
```
const transferredCanvasNames = 0;
for (let i of transferredCanvasNames) {
  console.log(i);
}
````
=>will fail with `transferredCanvasNames is not iterable`

---

I would suggest to go back to the following:
```
for (var i in transferredCanvasNames) {
  var name = transferredCanvasNames[i].trim();
```
(see the linked the pull request below)

### PR DESCRIPTION
**Pull request by boardend**
Partially reverts #17577 and fixes #17751

### CODE REVIEW

**Review by kripken — APPROVED** [review]
Thanks!

@sbc100 This is a regression from https://github.com/emscripten-core/emscripten/pull/17577 - perhaps we should check the other `for-of` loops there? I wasn't aware that `for-of` will throw when the input is e.g. `null`, but `for-in` does not.

File: src/library_pthread.js [hunk 1]
```diff
@@ -648,8 +648,8 @@ var LibraryPThread = {
 
     var offscreenCanvases = {}; // Dictionary of OffscreenCanvas objects we'll transfer to the created thread to own
     var moduleCanvasId = Module['canvas'] ? Module['canvas'].id : '';
-    for (var name of transferredCanvasNames) {
-      name = name.trim();
+    for (var i in transferredCanvasNames) {
```
Comment (kripken) [inline src/library_pthread.js:651]:
```suggestion
    // Note that transferredCanvasNames might be null (so we cannot do a for-of loop).  
    for (var i in transferredCanvasNames) {
```

**Comment by sbc100** [comment]
Can we add a test?

**Comment by kripken** [comment]
A test would be good, but as this is a fix for a regression, we should land this before making a new release. If adding a test would take time, we can maybe add the test later?

**Review by sbc100 — APPROVED** [review]

**Comment by kripken** [comment]
Ok, sounds good @sbc100 

@boardend I think merging in latest `main` should fix those CI errors.

**Pull request merged**